src/main/java/UseCases/PriceHistory/PriceHistoryUseCase.java

```diff
@@ -0,0 +1,173 @@
+        if (numDays == -1) {
+            return -1.0;
+        } else {
+            int i = 1;
+
+            double maxSoFar;
+            for(maxSoFar = this.item.getProductPrice(); i <= priceDataSize && i <= numDays; ++i) {
+                maxSoFar = Math.max(maxSoFar, (Double)this.item.getPriceHistoryData().get(priceDataSize - i));
+            }
+
+            return Double.valueOf(this.formatter.format(maxSoFar));
+        }
+    }
+
+    public double calculatePercentChangeFromDesiredPrice() {
+        return this.item.getProductDesiredPrice() == 0.0 ? -1.0 : Double.valueOf(this.formatter.format(this.item.getProductPrice() / this.item.getProductDesiredPrice() * 100.0));
+    }
+
+    public double calculatePercentChangeFromOriginalPrice() {
+        return (Double)this.item.getPriceHistoryData().get(0) == 0.0 ? -1.0 : Double.valueOf(this.formatter.format(this.item.getProductPrice() / (Double)this.item.getPriceHistoryData().get(0) * 100.0));
+    }
+
+    public double compareToAveragePrice(String timePeriod) {
+        int numDays = this.convertValidTimePeriodToDaysHelper(timePeriod);
+        if (numDays == -1) {
+            return -1.0;
+        } else {
+            return this.calculateAveragePrice(timePeriod) == 0.0 ? -1.0 : Double.valueOf(this.formatter.format(this.item.getProductPrice() / this.calculateAveragePrice(timePeriod) * 100.0));
+        }
+    }
+
+    public double compareToLowestPrice(String timePeriod) {
+        int numDays = this.convertValidTimePeriodToDaysHelper(timePeriod);
+        if (numDays == -1) {
+            return -1.0;
+        } else {
+            return this.calculateLowestPrice(timePeriod) == 0.0 ? -1.0 : Double.valueOf(this.formatter.format(this.item.getProductPrice() / this.calculateLowestPrice(timePeriod) * 100.0));
+        }
+    }
+
+    public double compareToHighestPrice(String timePeriod) {
+        int numDays = this.convertValidTimePeriodToDaysHelper(timePeriod);
+        if (numDays == -1) {
+            return -1.0;
+        } else {
+            return this.calculateHighestPrice(timePeriod) == 0.0 ? -1.0 : Double.valueOf(this.formatter.format(this.item.getProductPrice() / this.calculateHighestPrice(timePeriod) * 100.0));
+        }
+    }
+
+    public void generatePriceChart() {
+        DefaultCategoryDataset chartDataSet = new DefaultCategoryDataset();
+        ArrayList<Double> prices = this.item.getPriceHistoryData();
+        ArrayList<Date> dates = this.item.getPriceHistoryDates();
+        String datePattern = "MM-dd-yyyy";
+        SimpleDateFormat dateFormatter = new SimpleDateFormat(datePattern);
+        String series = "Date";
+
+        for(int i = 0; i < prices.size(); ++i) {
+            chartDataSet.addValue((Number)prices.get(i), series, dateFormatter.format((Date)dates.get(i)));
+        }
+
+        JFreeChart priceChart = ChartFactory.createLineChart("Price History", "Date", "Price", chartDataSet);
+
+        try {
```

Comment:
Could you move the exception logic into the gui?

Reply:
got it, took it out of the use case, thanks!